docs/notebooks/Training_visualizations.ipynb

```diff
@@ -0,0 +1,281 @@
+      "-22.4298221364\n"
+     ]
+    }
+   ],
+   "source": [
+    "# to get a metric value on a trained model\n",
+    "print(CoherenceMetric(corpus=training_corpus, coherence=\"u_mass\").get_value(model=model))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The four types of graphs which are plotted for LDA:\n",
+    "\n",
+    "**Coherence**\n",
+    "\n",
+    "Coherence is a measure used to evaluate topic models. A good model will generate coherent topics, i.e., topics         with high topic coherence scores. Good topics are topics that can be described by a short label based on the topic     terms they spit out. \n",
+    "\n",
+    "<img src=\"Coherence.gif\">\n",
+    "\n",
+    "Now, this graph along with the others explained below, can be used to decide if it's time to stop the training. We     can see if the value stops changing after some epochs and that we are able to get the highest possible coherence       of our model.  \n",
+    "\n",
+    "\n",
+    "**Perplexity**\n",
+    "\n",
+    "Perplexity is a measurement of how well a probability distribution or probability model predicts a sample. In LDA, topics are described by a probability distribution over vocabulary words. So, perplexity can be used to compare probabilistic models like LDA.\n",
+    "\n",
+    "<img src=\"Perplexity.gif\">\n",
+    "\n",
+    "For a good model, perplexity should be as low as possible.\n",
+    "\n",
+    "\n",
+    "**Topic Difference**\n",
+    "\n",
+    "Topic Diff calculates the distance between two LDA models. This distance is calculated based on the topics, by either using their probability distribution over vocabulary words (kullback_leibler, hellinger) or by simply using the common vocabulary words between the topics from both model.\n",
+    "\n",
+    "<img src=\"Diff.gif\">\n",
+    "\n",
+    "In the heatmap, X-axis define the Epoch no. and Y-axis define the distance between the identical topic from consecutive epochs. For ex. a particular cell in the heatmap with values (x=3, y=5, z=0.4) represent the distance(=0.4) between the topic 5 from 3rd epoch and topic 5 from 2nd epoch. With increasing epochs, the distance between the identical topics should decrease.\n",
+    "  \n",
+    "  \n",
+    "**Convergence**\n",
+    "\n",
+    "Convergence is the sum of the difference between all the identical topics from two consecutive epochs. It is basically the sum of column values in the heatmap above.\n",
+    "\n",
+    "<img src=\"Convergence.gif\">\n",
+    "\n",
+    "The model is said to be converged when the convergence value stops descending with increasing epochs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training Logs\n",
+    "\n",
+    "We can also log the metric values after every epoch to the shell apart from visualizing them in Visdom. The only change is to define `logger=\"shell\"` instead of `\"visdom\"` in the input callbacks."
```

Comment:
It can be done by using a loop to manually iterate over model and call metric classes at the end to store value:
```
model=LdaModel(passes=1)
perplexity=[]
for epoch in range(epochs):
    model.update(passes=1)
    pl = PerplexityMetric().get_value(model)
    perplexity.append(pl)
```
and sure, I'll add an example for this in notebook.